### DISCUSSION TITLE
Support env.local, et al. or support `--envfile` CLI option

### DISCUSSION THREAD

**Discussion by hparra** [amacneil/dbmate#354] [Feature Ideas]
dbmate does not support additional `.env` files, e.g. `.env.local` or `.env.development.local` due to https://github.com/joho/godotenv/issues/10

Possible solutions of varying completeness:
- Have [`loadDotEnv`](https://github.com/amacneil/dbmate/blob/519f186c8a3daec086608b2c21677c672eda56db/main.go#L215) also load `.env.local`
- Add `--envfile` option which specifies file for `godotenv.Load()`
- Use a fork of godotenv which implements https://github.com/joho/godotenv/issues/10

**Comment by msdrigg** [comment]
I also would like this issue supported

**Comment by amacneil** [comment]
Happy to accept a PR for this, I think it would only be a few lines of code

**Comment by msdrigg** [comment]
I haven't written any go code, but if this is still sitting open in a week or two, I may give it a try.

Also, if anyone does write a pr, I would definitely prefer the solution of
> Add `--envfile` option which specifies file for `godotenv.Load()`

**Comment by jahilliard** [comment]
I'd love this too.

**Comment by jahilliard** [comment]
Looks like something like `godotenv.Load(".env.local")` is supported.

**Comment by hamza512b** [comment]
I fixed it in this pr https://github.com/amacneil/dbmate/pull/491
Can @amacneil please look at it when you have time?

**Reply by dossy** [reply]
Thanks for submitting the PR, @hamza512b.  I've [reviewed](https://github.com/amacneil/dbmate/pull/491#pullrequestreview-1729930125) it, with requested changes.

**Comment by dossy** [comment]
For everyone who is interested in this feature, please take a look at #491.  Specifically, the behavior when the specified file does not exist, where we silently ignore and continue, and the behavior when multiple files are specified by repeating `--env-file <filename>`, where each file is applied in order and not stopping at the first found file.

The tests illustrate the expected behaviors:

https://github.com/amacneil/dbmate/blob/857ffed1f3332b596dacfcb21e38c352c45434c5/main_test.go#L63-L180

**Reply by dossy** [reply]
Update: #491 has been merged, and should land in a release after v2.8.0, probably v2.9.0.
